BoxContentSDK/BoxContentSDK/Helper/BOXRepresentationsHelper.m

```diff
@@ -0,0 +1,99 @@
+//
+//  BOXRepresentationsHelper.m
+//  BoxContentSDK
+//
+//  Created by Prithvi Jutur on 8/24/18.
+//  Copyright © 2018 Box. All rights reserved.
+//
+
+#import "BOXRepresentationsHelper.h"
+#import "BOXContentSDKConstants.h"
+
+@interface BOXRepresentationsHelper()
+
+///**
+// Setting a representation or list option will include availability of file with request representation information
+// */
+//@property (nonatomic, readwrite, strong) NSMutableOrderedSet *representationsRequested;
+
+/**
+ Ordered list of representation types, returns first representation type supported by the specified file type
+ */
+@property (nonatomic, readwrite, assign) BOOL matchSupportedRepresentation;
+@end
+
+@implementation BOXRepresentationsHelper
+
+- (void)setRepresentationRequestOptions:(NSArray *)representationOptions
+{
+    self.representationsRequested = [[NSMutableOrderedSet alloc] initWithArray:representationOptions];
+}
+
+- (NSString *)formatRepresentationRequestHeader
+{
+    // Only process valid API options.
+    if ([self.representationsRequested containsObject:@(BOXRepresentationRequestOriginal)]) {
+        [self.representationsRequested removeObject:@(BOXRepresentationRequestOriginal)];
+    }
+    
+    if ([self.representationsRequested count] == 0) {
+        return nil;
+    }
+    
+    __block NSString *representationFields = @"";
+    
+    if ([self.representationsRequested containsObject:@(BOXRepresentationRequestAllRepresentations)]) {
+        representationFields = @"[jpg?dimensions=320x320&paged=false][jpg?dimensions=1024x1024&paged=false][pdf,hls,mp4,mp3,jpg]";
+        [self.representationsRequested removeObject:@(BOXRepresentationRequestAllRepresentations)];
+    }
+    if ([self.representationsRequested count] == 0) {
+        return representationFields;
+    }
+    
+    representationFields = [representationFields stringByAppendingString:@"["];
+    
+    __block NSString *delimiter = @"],[";
+    if (self.matchSupportedRepresentation == YES) {
+        delimiter = @",";
+    }
+    
+    [self.representationsRequested enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
```

Comment:
This whole thing is crazy. Can we just map the set of numbers to an array of strings and `componentsJoinedByString:`?

Reply:
This isn't new, it just moved around